llvm/lib/Target/RISCV/RISCVAsmPrinter.cpp

```diff
@@ -517,12 +516,9 @@ void RISCVAsmPrinter::emitSled(const MachineInstr *MI, SledKind Kind) {
   // Assuming we're using JAL to jump to .tmpN, then we only need
   // (68 - 4)/2 = 32 NOPs for RV64 and (44 - 4)/2 = 20 for RV32. However, there
   // is a chance that we'll use C.JAL instead, so an additional NOP is needed.
-  const uint8_t NoopsInSledCount =
-      MI->getParent()->getParent()->getSubtarget<RISCVSubtarget>().is64Bit()
-          ? 33
-          : 21;
+  const uint8_t NoopsInSledCount = STI->is64Bit() ? 33 : 21;
 
-  OutStreamer->emitCodeAlignment(Align(4), &getSubtargetInfo());
```

Comment:
Is getSubtargetInfo() unused now? Can it be stripped?

Reply:
I think `getSubtargetInfo()` is fine. It's part of the AsmPrinter interface and just returns the non specialised SubtargetInfo - it's also just fetched from the MachineFunction.